### ISSUE TITLE
Bundler is not updating fastlane to the latest build

### ISSUE THREAD

**Issue by Vkt0r** [fastlane/fastlane#11607]
### New Issue Checklist

- [ ] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Some time ago I reported this issue in Bundler https://github.com/bundler/bundler/issues/6065 regarding `bundle` was not updating fastlane to the latest version. If you follow the issue comments a fastlane PR was referenced https://github.com/fastlane/fastlane/pull/10168. 

At that time the issue with `bundler` was a downgrade of fastlane there in order to support an update of `public_suffix`. As `bundler` doesn't have a rule for which of two conflicting updates should "win" when running `bundle update` according to @greysteil told me last time.

Running `bundle update fastlane` doesn't make any difference at all. Below you cand find my `Gemfile.lock` in case you need it.

<details><summary><strong>Gemfile.lock</strong> </summary>

```
GEM
  remote: https://rubygems.org/
  specs:
    CFPropertyList (2.3.6)
    addressable (2.5.2)
      public_suffix (>= 2.0.2, < 4.0)
    babosa (1.0.2)
    claide (1.0.2)
    claide-plugins (0.9.2)
      cork
      nap
      open4 (~> 1.3)
    colored (1.2)
    colored2 (3.1.2)
    commander-fastlane (4.4.5)
      highline (~> 1.7.2)
    cork (0.3.0)
      colored2 (~> 3.1)
    danger (5.5.6)
      claide (~> 1.0)
      claide-plugins (>= 0.9.2)
      colored2 (~> 3.1)
      cork (~> 0.1)
      faraday (~> 0.9)
      faraday-http-cache (~> 1.0)
      git (~> 1)
      kramdown (~> 1.5)
      no_proxy_fix
      octokit (~> 4.7)
      terminal-table (~> 1)
    danger-junit (0.7.4)
      danger (> 2.0)
      ox (~> 2.0)
    danger-prose (2.0.5)
      danger
    danger-swiftlint (0.12.1)
      danger
      rake (> 10)
      thor (~> 0.19)
    danger-xcov (0.4.1)
      danger (>= 2.1)
      xcov (>= 1.1.2)
    declarative (0.0.10)
    declarative-option (0.1.0)
    domain_name (0.5.20170404)
      unf (>= 0.0.5, < 1.0.0)
    dotenv (2.2.1)
    excon (0.60.0)
    faraday (0.13.1)
      multipart-post (>= 1.2, < 3)
    faraday-cookie_jar (0.0.6)
      faraday (>= 0.7.4)
      http-cookie (~> 1.0.0)
    faraday-http-cache (1.3.1)
      faraday (~> 0.8)
    faraday_middleware (0.12.2)
      faraday (>= 0.7.4, < 1.0)
    fastimage (2.1.1)
    fastlane (2.75.1)
      CFPropertyList (>= 2.3, < 3.0.0)
      addressable (>= 2.3, < 3.0.0)
      babosa (>= 1.0.2, < 2.0.0)
      bundler (>= 1.12.0, < 2.0.0)
      colored
      commander-fastlane (>= 4.4.5, < 5.0.0)
      dotenv (>= 2.1.1, < 3.0.0)
      excon (>= 0.45.0, < 1.0.0)
      faraday (~> 0.9)
      faraday-cookie_jar (~> 0.0.6)
      faraday_middleware (~> 0.9)
      fastimage (>= 2.1.0, < 3.0.0)
      gh_inspector (>= 1.0.1, < 2.0.0)
      google-api-client (>= 0.13.1, < 0.14.0)
      highline (>= 1.7.2, < 2.0.0)
      json (< 3.0.0)
      mini_magick (~> 4.5.1)
      multi_json
      multi_xml (~> 0.5)
      multipart-post (~> 2.0.0)
      plist (>= 3.1.0, < 4.0.0)
      public_suffix (~> 2.0.0)
      rubyzip (>= 1.1.0, < 2.0.0)
      security (= 0.1.3)
      slack-notifier (>= 1.3, < 2.0.0)
      terminal-notifier (>= 1.6.2, < 2.0.0)
      terminal-table (>= 1.4.5, < 2.0.0)
      tty-screen (>= 0.6.3, < 1.0.0)
      tty-spinner (>= 0.7.0, < 1.0.0)
      word_wrap (~> 1.0.0)
      xcodeproj (>= 1.5.2, < 2.0.0)
      xcpretty (>= 0.2.4, < 1.0.0)
      xcpretty-travis-formatter (>= 0.0.3)
    gh_inspector (1.0.3)
    git (1.3.0)
    google-api-client (0.13.6)
      addressable (~> 2.5, >= 2.5.1)
      googleauth (~> 0.5)
      httpclient (>= 2.8.1, < 3.0)
      mime-types (~> 3.0)
      representable (~> 3.0)
      retriable (>= 2.0, < 4.0)
    googleauth (0.6.2)
      faraday (~> 0.12)
      jwt (>= 1.4, < 3.0)
      logging (~> 2.0)
      memoist (~> 0.12)
      multi_json (~> 1.11)
      os (~> 0.9)
      signet (~> 0.7)
    highline (1.7.10)
    http-cookie (1.0.3)
      domain_name (~> 0.5)
    httpclient (2.8.3)
    json (2.1.0)
    jwt (2.1.0)
    kramdown (1.16.2)
    little-plugger (1.1.4)
    logging (2.2.2)
      little-plugger (~> 1.1)
      multi_json (~> 1.10)
    memoist (0.16.0)
    mime-types (3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2016.0521)
    mini_magick (4.5.1)
    multi_json (1.13.1)
    multi_xml (0.6.0)
    multipart-post (2.0.0)
    nanaimo (0.2.3)
    nap (1.1.0)
    no_proxy_fix (0.1.2)
    octokit (4.8.0)
      sawyer (~> 0.8.0, >= 0.5.3)
    open4 (1.3.4)
    os (0.9.6)
    ox (2.8.2)
    plist (3.4.0)
    public_suffix (2.0.5)
    rake (12.3.0)
    representable (3.0.4)
      declarative (< 0.1.0)
      declarative-option (< 0.2.0)
      uber (< 0.2.0)
    retriable (3.1.1)
    rouge (2.0.7)
    rubyzip (1.2.1)
    sawyer (0.8.1)
      addressable (>= 2.3.5, < 2.6)
      faraday (~> 0.8, < 1.0)
    security (0.1.3)
    signet (0.8.1)
      addressable (~> 2.3)
      faraday (~> 0.9)
      jwt (>= 1.5, < 3.0)
      multi_json (~> 1.10)
    slack-notifier (1.5.1)
    terminal-notifier (1.8.0)
    terminal-table (1.8.0)
      unicode-display_width (~> 1.1, >= 1.1.1)
    thor (0.20.0)
    tty-cursor (0.5.0)
    tty-screen (0.6.4)
    tty-spinner (0.8.0)
      tty-cursor (>= 0.5.0)
    uber (0.1.0)
    unf (0.1.4)
      unf_ext
    unf_ext (0.0.7.4)
    unicode-display_width (1.3.0)
    word_wrap (1.0.0)
    xcodeproj (1.5.4)
      CFPropertyList (~> 2.3.3)
      claide (>= 1.0.2, < 2.0)
      colored2 (~> 3.1)
      nanaimo (~> 0.2.3)
    xcov (1.3.1)
      fastlane (>= 2.19.2, < 3.0.0)
      multipart-post
      slack-notifier (~> 1.3)
      terminal-table
      xcodeproj (~> 1.4)
    xcpretty (0.2.8)
      rouge (~> 2.0.7)
    xcpretty-travis-formatter (1.0.0)
      xcpretty (~> 0.2, >= 0.0.7)

PLATFORMS
  ruby

DEPENDENCIES
  danger
  danger-junit
  danger-prose
  danger-swiftlint
  danger-xcov
  fastlane
  xcov

BUNDLED WITH
   1.16.1
```

</details>
</br>

So I think this issue is back again 😢.

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.2                                     |
| Ruby                        | 2.4.1                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.14.3 (Apple Git-98)           |
| Installation Source         | ~/.rvm/gems/ruby-2.4.1/bin/fastlane         |
| Host                        | Mac OS X 10.13.2 (17C205)                   |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.1/lib                |
| OpenSSL Version             | OpenSSL 1.1.0g  2 Nov 2017                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.75.1  | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| bundler                   | 1.16.1       |
| io-console                | 0.4.6        |
| rake                      | 12.3.0       |
| CFPropertyList            | 2.3.6        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| cork                      | 0.3.0        |
| nap                       | 1.1.0        |
| open4                     | 1.3.4        |
| claide-plugins            | 0.9.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| multipart-post            | 2.0.0        |
| faraday                   | 0.13.1       |
| faraday-http-cache        | 1.3.1        |
| git                       | 1.3.0        |
| kramdown                  | 1.16.2       |
| no_proxy_fix              | 0.1.2        |
| sawyer                    | 0.8.1        |
| octokit                   | 4.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| danger                    | 5.5.6        |
| ox                        | 2.8.2        |
| danger-junit              | 0.7.4        |
| danger-prose              | 2.0.5        |
| thor                      | 0.20.0       |
| danger-swiftlint          | 0.12.1       |
| dotenv                    | 2.2.1        |
| excon                     | 0.60.0       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.12.2       |
| fastimage                 | 2.1.1        |
| gh_inspector              | 1.0.3        |
| jwt                       | 2.1.0        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.13.1       |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| google-api-client         | 0.13.6       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.4.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.8.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.4        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| xcpretty-travis-formatter | 1.0.0        |
| xcov                      | 1.3.1        |
| danger-xcov               | 0.4.1        |
</details>

*generated on:* **2018-01-18**
</details>

**Comment by Vkt0r** [comment]
With the new version `2.77.1` I was able to update `fastlane` using Bundler, nevertheless, I would like to know if you guys found something related the downgrades of dependencies you have that cause at some point the impossibility of update `fastlane` regarding the Bundler policies ?

**Comment by sprint84** [comment]
I'm having the same issue trying to update fast lane `2.77.1` to `2.79.0`
Backtrace prints:

`RuntimeError: Unable to find a spec satisfying fastlane (>= 0) in the set. Perhaps the lockfile is corrupted?`

**Comment by Vkt0r** [comment]
@sprint84 I think this different 🤔. Did you tried to remove your `Gemfile.lock` and run `bundle install`?

**Comment by taquitos** [comment]
@Vkt0r you might need to `rvm implode` and start fresh again, although I'm not an rvm expert.

**Comment by Vkt0r** [comment]
@taquitos No, I wasn't using `rvm` to manage the Ruby versions, I check it and it was regarding the Bundler issue I commented above, a downgrade problem it seems. Luckily it was fixed after `2.77.1`.

**Comment by joshdholtz** [comment]
Seems like things were fixed so closing this! Comment back if this happens again 😊
